### PR TITLE
Writeinit fp

### PR DESCRIPTION
**Pull request by joakim-hove**
Rewrite writeInit() to use `FieldPropsManager`

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/WriteInit.cpp [hunk 1]
```diff
@@ -363,10 +368,19 @@ namespace {
...
                                             const ::Opm::EclipseGrid&       grid,
                                             WriteVector&&                   write)
     {
         for (const auto& prop : propList) {
+#ifdef ENABLE_3DPROPS_TESTING
+            if (! fp.has<double>(prop.name))
+                continue;
+
+            auto data = fp.get<double>(prop.name);
+            auto defaulted = fp.defaulted<double>(prop.name);
```
Comment (bska) [inline src/opm/output/eclipse/WriteInit.cpp]: These should be `fp.has<T>()`, `fp.get<T>()`, and `fp.defaulted<T>()`.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/WriteInit.cpp [hunk 2]
```diff
@@ -316,19 +317,31 @@ namespace {
...
+        std::vector<double> porv(grid.getCartesianSize());
+        std::size_t active_index = 0;
+        for (auto nGlob    = porv.size(),
+                 globCell = 0*nGlob; globCell < nGlob; ++globCell) {
+            if (grid.cellActive(globCell)) {
+                porv[globCell] = active_porv[active_index];
+                active_index++;
+            } else {
+                porv[globCell] = 0.0;
+            }
```
Comment (bska) [inline src/opm/output/eclipse/WriteInit.cpp]: You don't actually need this `else` branch.  Constructing the `porv` vector above will implicitly assign `0.0` to all elements.  In short, this loop is equivalent to
```C++
for (...) {
    if (grid.cellActive(globCell)) {
        // Assign porv
    }
}
```
On the other hand, I don't think you need to loop the global cells at all.  You should be able to write this along the lines of
```C++
std::vector<double> porv(grid.getCartesianSize());  // or porv(grid.getCartesianSize(), 0.0)
std::size_t active_index = 0;
for (const auto& pv : es.fieldProps().porv()) {
    porv[grid.getGlobalIndex(active_index++)] = pv;
}
```

Comment (joakim-hove) [reply src/opm/output/eclipse/WriteInit.cpp]: Thank you for the suggestions - I will probably implement them.

**Comment by joakim-hove** [comment]
> [...] otherwise this should go in when the build tests are good.

Thank you for the review.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Observe the last commit in this pr which will effectively make the comparison of INIT files oblivious to keyword reordering. This is temporary, and will be reverted quite soon - as agreed here: #1305 - but while the ordering of the depends on whether the `ENABLE_3DPROPS_TESTING`flag is set or not (it is by default OFF, but is on on Jenkins) it will be very unstable to fail on keyword reordering.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=428 please

**Comment by joakim-hove** [comment]
jenkins build this opm-grid=428 opm-simulators=2216 please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=375 please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**